js/fullWidthTreeView.js

```diff
@@ -218,12 +235,39 @@
     // a node is hovered to make it appear after node changes. It must
     // use the #fullwidth-treeview container to allow a higher
     // height than the node in multiple lines tooltips
+    var syncedParents = {};
     var hoverNodeListener = function (e, data)
     {
+      // Do nothing if node is disabled
+      if ($fwTreeView.jstree("is_disabled", data.node.id)) {
+        return;
+      }
+
+      // Configure tooltip
       $('a.jstree-anchor').tooltip({
         delay: 250,
         container: '#fullwidth-treeview'
       });
+
+      // Check sync of treeview data
+      var parent = data.node.parent;
+      var parentNode = $fwTreeView.jstree("get_json", parent);
+
+      if (parent != "#" && !(parent in syncedParents) && ("href" in parentNode.a_attr)) {
+        syncedParents[parent] = true;
+        commandNodeAndChildren($fwTreeView, parent, "disable_node");
+
+        var url = parentNode.a_attr.href + '/informationobject/fullWidthTreeViewSync';
+
+        $.get(url, function (response)
+        {
+          if (response["repaired"]) {
+            $fwTreeView.jstree("refresh_node", parent);
+          }
+
+          commandNodeAndChildren($fwTreeView, parent, "enable_node");
+        });
+      }
```

Comment:
I'd try not to sync on hover as it may happen a lot of times, if possible I'd do it when ready and when the node is selected/loaded.

Reply:
There's a hash `syncInitiated`, with node parent ID as the key, that gets set to `true` once a hover has triggered a sync check. Unless it's later set to `false` (or anything other than `true`) a sync check won't happen on subsequent hovers over the children of the same parent.

I considered the expanding of a parent's children as a potential event type to use to trigger the sync check, but in some cases a user will expand a parent's children but not interact with the children (making an immediate sync not necessary). By triggering a sync check only when a user hovers over a node's children there's no delay to get results when opening a node.